_posts/multirepo_comments.adoc

```diff
@@ -0,0 +1,96 @@
+# The Dangers of Multirepo
+
+In this post I'm going to make the case for a somewhat unpopular opinion - using multiple git repos (AKA multirepo) is unnecessary and dangerous in many cases. A single git repo (AKA monorepo) is simpler and satisfies the requirements of many projects.
+
+## The Story
+
+I've seen this happening multiple times: you and a few teammates are starting a new project. The project is some complicated system that comprises several processes. You `git init` a new repo for each process without thinking about it too much, it feels natural. 
+
+As the system evolves, the processes start talking to each other over RPC, and since the processes use different programming languages, you use something like `protobuf` or `thrift`. Since the `.proto/.thrift` files are used by multiple components, they don't belong in any of the repos you created, so you create a new git repo and stick the shared API stuff in it.
+
+Since you're a great software engineer you always write unit tests for your code, and that fits snugly in your component's repo. A few weeks later, the system is a bit more mature and you want to start adding integration testing between components and system level testing to test the entire thing. But that doesn't really fit in any component's repo, and it's certainly not a shared-api thingy, so you create another repo for system testing.
+
+Later on you realize that two of the components can share some piece of infrastructure code, so you create a library out of it and put that in its own repo.
+
+Then you want some "magic script" to build your entire system from scratch and produce a shipable product, so you write it up. But where do you put it? Doesn't seem to fit in any of the repos you created... So you create a new repo for build stuff.
+
+Without noticing, you've just created ~10 git repositories for a single product. Welcome to MultiRepo Hell.
+
+While the above example sound extreme, I've actually seen this happen more than once. We'll now examine why this is a terrible way to go.
+
```

Comment:
Let's number your point so that we can keep track.

1. I was wandering if it's better to give an example or talk in general terms, but couldn't make my mind. Do you think an example is easier to follow?
2. How can I know if anyone saw this before or not?
3. What I'm trying to suggest is that many projects "choose" multirepo without thinking about it too much, it sort of "just happens". Is that not clear from the introduction?
4. I agree people usually don't read things thoroughly. What changes can I make to deliver "the story" without burdening people with too much text?